src/windows/scanner.cc

```diff
@@ -220,15 +220,23 @@ ScanMountpoints(drivelist::com::Connection *const connection,
     if (FAILED(result))
       return InterpretHRESULT(result);
 
-    BOOL isProtected;
-    result = drivelist::volume::GetReadOnlyFlag(path[0], &isProtected);
+    // Turns out a disk can be writable while its volumes can be read-only
+    // and viceversa, so we need to check the writable capabilities of
+    // both the disk and its volumes
+    BOOL writable;
+    result = drivelist::volume::IsDiskWritable(path[0], &writable);
     if (FAILED(result))
       return InterpretHRESULT(result);
+    if (writable && hasFilesystem) {
+      result = drivelist::volume::IsVolumeWritable(path[0], &writable);
+      if (FAILED(result))
+        return InterpretHRESULT(result);
+    }
 
     drivelist::mountpoint_s mountpoint;
     mountpoint.path = path;
     mountpoint.disk = "\\\\.\\PHYSICALDRIVE" + std::to_string(number);
-    mountpoint.readonly = isProtected;
+    mountpoint.readonly = !writable;
     mountpoint.system = path[0] == systemDriveLetter;
     mountpoint.hasFilesystem = hasFilesystem;
```

Comment:
Both `IOCTL_DISK_IS_WRITABLE` and `GetVolumeInformation()` take letters as arguments, but one of them returns the writable state of the disk, and the other returns the writable state of the volume. There doesn't seem to be any Windows function to determine the writable state of a drive.

Also, it looks like despite having the same identifier, a volume and a disk are different things, and each have their own copy of attributes such as this one.

I guess these kinds of awkwardness are what make the Windows APIs so confusing. Check https://stackoverflow.com/questions/44703463/getvolumeinformation-not-reporting-file-read-only-volume-for-a-locked-sd-card/44706760 for some more details.